### ISSUE TITLE
make update_python_requirements.bsh support binary only packages not on pypi

### ISSUE THREAD

**Issue by scott-vsi** [VisionSystemsInc/vsi_common#23]
With the update_python_requirements.bsh:PIP_BINARY_ONLY env variable, it is possible to specify which packages from pypi should be treated as binary only. I suggest extending this to support packages from other locations (e.g., PyTorch).

**Comment by andyneff** [comment]
## tl;dr

This is more synonymous to how some packages need numpy installed before installing. So it's up to the user to install torch first before calling the update script. Perhaps adding it to a virtualenv like you would numpy would work?

```dockerfile
RUN virtualenv /opt/venv; \
    PS1= source /opt/venv/bin/activate; \
    pip install --no-cache-dir setuptools==35.0.2 pip==9.0.1; \
    pip install --no-cache-dir numpy==1.12.1
```

---

`PIP_BINARY_ONLY` is a `pip` feature, not vsi_common feature.

The intended use of `update_python_requirements` is to update the python requirements.

So if I said I need "scipy=0.19.1" as a requirement, then I want to update all the requirements to the latest versions of the package, say "scipy=1.0.0". This *update* process is done by asking pypi via pip what are the latest versions.

For packages such as pytorch, that are not download from a repository like pypi, rather a simple url, asking the question "What is the newest version" does not compute. This is why a package like pytorch's url should be added to the custom requirements file.

```
http://download.pytorch.org/whl/cu80/torch-0.3.1-cp36-cp36m-linux_x86_64.whl 
```

---

Now packages like `torchvision` are a problem, as the are in pypi, and require torch, which is not. This is essentially a broken package. For this reason, you can not run

```
pip download -d /tmp/foo torchvision
```

as some torch-0.1.2.post1.tar.gz release will be [downloaded](https://pypi.python.org/simple/torch/), and have setup.py run, which subsequently fails by design.

Without having manually installed torch first, there's just no other way, so barring a better solution, torchvision would need to be added as a custom package too, as it is indeed **broken**

The correct thing to do is to have `torchvision` [remove the requirement](https://github.com/pytorch/vision/blob/v0.2.0/setup.py#L35) for `torch`
